src/core/core.tooltip.js

```diff
@@ -463,6 +463,25 @@ function getBackgroundPoint(vm, size, alignment, chart) {
 	};
 }
 
+/**
+ * Align pt.x to match given align (and using xPadding)
```

Comment:
Agree.

Personally, I like jsdocs for parameter hints (even for private stuff). But if documentation is generated based on those, then said documentation would be polluted. 

TBH I thought this would be relocated (to helpers) initially, but was not considering public api support burden.

Reply:
> ... jsdocs for parameter hints

For basic types it could help, but for objects, we would need to detail the whole structure:

```javascript
// @param {Object} vm - rectangle text should be aligned in {x, y, width, height, xPadding}
```

What are `x, y, width, height, xPadding`?

```javascript
// @param {Object} vm - the rectangle model
// @param {number} vm.x - the rectange x position
// @param {number} vm.y - the rectange y position
// @param {number} vm.width - the rectange width
// @param {number} vm.height - the rectange height
// @param {number} vm.xPadding - the horizontal padding
```

And it could be even more verbose, e.g.

```javascript
/**
 * Generate a set of logarithmic ticks
 * @param generationOptions the options used to generate the ticks
 * @param dataRange the range of the data
 * @returns {Array<Number>} array of tick values
 */
function generateTicks(generationOptions, dataRange) {
```

A comment like this is useless IMO because the description doesn't add any value and I can't figure out how to call this method only reading the jsdoc.

> ... then said documentation would be polluted

This can be controlled by excluding privates from the document, but it's more about the maintenance and usefulness of these methods which change a lot (if I have to change this method signature, I will simply delete this whole comment). And honestly, I don't want to spend time to write tons of (potentially never read) comments for methods that I will have to change or remove in the next days.

> ... I thought this would be relocated (to helpers) initially

We should consider all new methods (including helpers) to be private unless it's explicitly requested and it makes sense. In this case we will have to commit on the API to make it consistent and durable. I don't think this method needs to be public and isn't generic enough to be public.

> **me:** ... adds noise ... time consuming to write, read and maintain

See [4463](/chartjs/Chart.js/pull/4463/files) as an example of over-documented internals.

**To sum up:** I would stop jsdoc for internals and write much better jsdoc for the public API.